### PR TITLE
Cigarette ecs

### PR DESCRIPTION
**Pull request by Ygg01**
Removes old Component-based `StorageCounter` replaces it with Systems. 

Also renames old `ItemCounterComponent` into `ItemMapperComponent` to distinguish between these two.
When making the Belt tool visualizer I intended to unify them, but on @mirrorcult suggestion, I'll keep them separated.

Not interesting for players, so no changelog.

### CODE REVIEW

File: Content.Server/Storage/EntitySystems/ItemCounterSystem.cs [hunk 1]
```diff
@@ -37,5 +39,26 @@ public class ItemCounterSystem : SharedItemCounterSystem
             showLayers = new List<string>();
             return false;
         }
+
+        protected override bool TryGetCount(ContainerModifiedMessage msg, ItemCounterComponent itemCounter,
```
Comment (mirrorcult) [inline Content.Server/Storage/EntitySystems/ItemCounterSystem.cs:43]: why bother with an out param, i would just return 0 if theres no server storage or no items

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by mirrorcult** [comment]
if you want to unity itemmapper/itemcounter and you think that would lead to cleaner code, go for it, my opinion on that isn't final

**Comment by mirrorcult** [comment]
some yaml still has some removed components

**Comment by Ygg01** [comment]
> some yaml still has some removed components

I ran the YAML Linter and it showed no errors, if that's what you mean.

**Comment by PaulRitter** [comment]
> > some yaml still has some removed components
> 
> I ran the YAML Linter and it showed no errors, if that's what you mean.

Linter says no
image

**Comment by Ygg01** [comment]
> Linter says no
> image

Are you just parroting the Linter or did you actually check it?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
> 
> 
> > Linter says no
> > image
> 
> Are you just parroting the Linter or did you actually check it?
> 
> EDIT: Those lines don't exist on my branch

You need to merge in master my man.

**Comment by Ygg01** [comment]
> You need to merge in master my man.

This is a helpful comment. Saying fix isn't as much.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**